### ISSUE TITLE
Unreleased gradeable data leak

### ISSUE THREAD

**Issue by lavalleeale** [Submitty/Submitty#11101]
**Describe the bug**
Future gradeable ids can be leaked by going to their leaderboard page, and depending on the error the student can tell whether there is a gradeable with that id or not.

**Expected behavior**

There should be the same error if the gradeable does not exist or if it is not yet released to students.

**To Reproduce**
1. Visit  http://localhost:1511/courses/f24/sample/gradeable/bulk_upload_test/leaderboard (unreleased gradeable)
2. See error "No leaderboard exists for this gradeable"
3. Visit http://localhost:1511/courses/f24/sample/gradeable/bulk_upload_tes/leaderboard (non-existent gradeable)
4. See different error "Invalid gradeable id"

**Comment by williamschen23** [comment]
dupe of #11100

**Comment by lavalleeale** [comment]
> dupe of #11100 ? maybve not?

I don't think so, that is about seeing the leaderboard, this is for gradeables that do not have a leaderboard configured. They might still be solved with the same PR but are distinct issues to solve.

**Comment by urek69mazino** [comment]
even after searching all the file in the grading and autograding folders , i am not able to find the file that contains the following error /exception , kindly guide me , which folders to explore ?

**Comment by lavalleeale** [comment]
> even after searching all the file in the grading and autograding folders , i am not able to find the file that contains the following error /exception , kindly guide me , which folders to explore ?

They should be on the top of the screen of the student attempting to view the gradeables

**Comment by williamschen23** [comment]
> even after searching all the file in the grading and autograding folders , i am not able to find the file that contains the following error /exception , kindly guide me , which folders to explore ?

leaderboard controller

**Comment by williamschen23** [comment]
This issue seems like it has to do with a broader access control system, so removing good first issue. https://github.com/Submitty/Submitty/blob/c7f3763ebf0bd3df2b5f0a8081f80a08913a826f/site/app/controllers/student/SubmissionController.php#L37-L47
